Tests/UnitTestsParallelizable/Views/TextFieldTests.cs

```diff
@@ -0,0 +1,555 @@
+﻿using System.Text;
+
+namespace Terminal.Gui.ViewsTests;
+
+public class TextFieldTests
+{
+    [Fact]
+    public void Cancel_TextChanging_ThenBackspace ()
+    {
+        var tf = new TextField ();
+        tf.SetFocus ();
+        tf.NewKeyDownEvent (Key.A.WithShift);
+        Assert.Equal ("A", tf.Text);
+
+        // cancel the next keystroke
+        tf.TextChanging += (s, e) => e.Cancel = e.NewValue == "AB";
+        tf.NewKeyDownEvent (Key.B.WithShift);
+
+        // B was canceled so should just be A
+        Assert.Equal ("A", tf.Text);
+
+        // now delete the A
+        tf.NewKeyDownEvent (Key.Backspace);
+
+        Assert.Equal ("", tf.Text);
+    }
+
+    [Fact]
+    public void HistoryText_IsDirty_ClearHistoryChanges ()
+    {
+        var text = "Testing";
+        var tf = new TextField { Text = text };
+        tf.BeginInit ();
+        tf.EndInit ();
+
+        Assert.Equal (text, tf.Text);
+        tf.ClearHistoryChanges ();
+        Assert.False (tf.IsDirty);
+
+        Assert.True (tf.NewKeyDownEvent (Key.A.WithShift));
+        Assert.Equal ($"{text}A", tf.Text);
+        Assert.True (tf.IsDirty);
+    }
+
+    [Fact]
+    public void Space_Does_Not_Raise_Selected ()
+    {
+        TextField tf = new ();
+
+        tf.Selecting += (sender, args) => Assert.Fail ("Selected should not be raied.");
+
+        Application.Top = new ();
```

Comment:
Tests that access `Application` cannot be in Parallelizable.

Please refactor this to not use Application.Top. Or, if that's not possible, move it back to `UnitTests`.

In this case, I don't think there's any reason to use `Application`. Just create a superview and call `superView.RaiseNewKeyDownEvent`.